lib/semantics/check-do-concurrent.cc

```diff
@@ -160,25 +192,38 @@ class DoConcurrentEnforcement {
     }
     return false;
   }
+  bool addName(const std::optional<parser::Name> &nm) {
+    if (nm.has_value())
+      names_.insert(nm.value().ToString());
```

Comment:
There is no need to create a string here. Instead you could make names_ a set of parser::CharBlock. There here insert `nm.value().source`.

Reply:
Thanks, I have made the suggested change.